### PR TITLE
[WIP] Reactify 

### PR DESCRIPTION
**Pull request by microbit-grace**
**TODO:** 
- [x] Home page
- [ ] Resources pages

Connection flow
- [x] Web USB flashing
- [ ] Bluetooth connection logic
- [ ] Connection states and hooks
- [ ] Bluetooth and usb compatibility hook (also sub in where it is currently mocked in dialogs)
- [x] Connection flow dialogs

Others
- [ ] Add data page
- [ ] Train model page
- [ ] Test model page
- [ ] Replace support links where there is TODO
- [ ] Reconnect flow and dialogs

### CODE REVIEW

File: src/components/SelectMicrobitUsbDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+export interface SelectMicrobitDialogProps
+  extends Omit<ConnectContainerDialogProps, "children" | "headingId"> {}
+
+const SelectMicrobitUsbDialog = ({ ...props }: SelectMicrobitDialogProps) => {
+  const intl = useIntl();
+
+  return (
+    <ConnectContainerDialog headingId="connectMB.webPopup" {...props}>
+      <Box position="relative" width={"100%"} alignSelf={""}>
```
Comment (microbit-matt-hillsdon) [inline src/components/SelectMicrobitUsbDialog.tsx:26]: {""} seems odd

```suggestion
      <Box position="relative" width={"100%"} alignSelf={""}>
```

File: src/components/SettingsMenu.tsx [hunk 2]
```diff
@@ -1,17 +1,11 @@
...
-} from "@chakra-ui/react";
+import { IconButton, Menu, MenuButton, MenuList } from "@chakra-ui/react";
 import { RiSettings2Line } from "react-icons/ri";
 import { useIntl } from "react-intl";
 import LanguageMenuItem from "./LanguageMenuItem";
 import { useRef } from "react";
 
 interface SettingsMenuProps {
-  variant?: ThemeTypings["components"]["Menu"]["variants"];
+  variant?: string;
```
Comment (microbit-matt-hillsdon) [inline src/components/SettingsMenu.tsx:8]: Did the theme type cause issues?

File: src/connection-flow.ts [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+    { stage: ConnStage.WebUsbFlashingTutorial, type: RadioRemote },
+    { stage: ConnStage.ConnectBattery, type: RadioRemote },
+    { stage: ConnStage.ConnectCable, type: RadioBridge },
+    { stage: ConnStage.WebUsbFlashingTutorial, type: RadioBridge },
+  ];
+};
+
+const getItemIdx = (item: object, arr: object[]) => {
+  return arr.map((a) => JSON.stringify(a)).indexOf(JSON.stringify(item));
+};
```
Comment (microbit-matt-hillsdon) [inline src/connection-flow.ts:161]: This seems a bit unusual - might be worth finding another way.

File: src/deployment/misc.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+/**
+ * (c) 2021, Micro:bit Educational Foundation and contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+// Miscellaneous items that haven't been properly integrated into the branding.
+export const backgroundColorTerm = "#333333";
+export const defaultCodeFontSizePt = 14;
```
Comment (microbit-matt-hillsdon) [inline src/deployment/misc.ts:9]: Some of these feel likely to be unused here.

**Comment by microbit-matt-hillsdon** [comment]
I think we should take this opportunity to decide how best to share the WebUSB-related code with Python and see if we incorporate Web Bluetooth in a similar way.

File: src/components/SelectMicrobitUsbDialog.tsx (hunk 1 above)
Comment (microbit-grace) [reply src/components/SelectMicrobitUsbDialog.tsx]: https://github.com/microbit-foundation/ml-trainer/pull/267/commits/d6274b15330beb03aafe3f63f85e38e38c2f44c9

File: src/components/SettingsMenu.tsx (hunk 2 above)
Comment (microbit-grace) [reply src/components/SettingsMenu.tsx:8]: Yes!
Screenshot 2024-07-04 at 13 13 17

File: src/deployment/misc.ts (hunk 4 above)
Comment (microbit-grace) [reply src/deployment/misc.ts]: I have removed misc.ts here -> https://github.com/microbit-foundation/ml-trainer/pull/267/commits/39e456b54400e5b1441304b1eba5180bc5839155

File: src/components/SettingsMenu.tsx (hunk 2 above)
Comment (microbit-grace) [reply src/components/SettingsMenu.tsx]: I have updated it here: https://github.com/microbit-foundation/ml-trainer/pull/267/commits/8c7953e38673c3599bff1ec60a26fd55e921fe2c
Strange that I don't get that typescript error in the Python Editor 🤔

Comment (microbit-matt-hillsdon) [reply src/components/SettingsMenu.tsx]: Fixed in eda9b836d612af5e1440801d1cce59510145410d

**Pull request merged**